doozer/doozerlib/backend/konflux_client.py

```diff
@@ -712,3 +714,84 @@ def _inner():
                     raise
 
         return await exectools.to_thread(_inner)
+
+    async def wait_for_release(self, release_name: str, namespace: Optional[str] = None,
+                               overall_timeout_timedelta: Optional[datetime.timedelta] = None) -> resource.ResourceInstance:
+        """
+        Wait for a Release to complete.
+
+        :param release_name: The name of the Release.
+        :param namespace: The namespace of the Release.
+        :param overall_timeout_timedelta: Maximum time to wait for release to complete before exiting (defaults to 5 hour)
+        :return: The Release ResourceInstance
+        """
+        if overall_timeout_timedelta is None:
+            overall_timeout_timedelta = datetime.timedelta(hours=5)
+
+        namespace = namespace or self.default_namespace
+        api = await self._get_api(API_VERSION, KIND_RELEASE)
+
+        if self.dry_run:
+            await asyncio.sleep(3)
```

Comment:
No need to wait during a dry run?

Reply:
I just followed the pattern we set in `wait_for_pipelinerun` above this :)